### PR TITLE
Introduced Sync package 

### PR DESCRIPTION
**Pull request by iadityanath8**
In this pull request i have introduced the sync mechanism as for now Mutex and abstracted pthread api in linux now we can easily 
lock, unlock as well as we can also set the attribute in the mutex also this package has been inspired by the design of thread module and this package uses that design in sync package also

### CODE REVIEW

**Review by mustafaquraish — CHANGES_REQUESTED** [review]
Thanks a lot for the PR 🙂 Other than some style issues, it looks great! 

Left a bunch of comments here, please revise them. 

Aside from style, bigger picture: let's put this stuff in `std::thread`. I don't see a need for this to be a separate thing.

File: std/sync.oc [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import std::thread::{Thread}
+import std::libc::{errno,strerror,exit}
+
+const EXIT_FAILURE:i32 extern
```
[the remark this answers is not in the document]
Comment (iadityanath8) [reply std/sync.oc:4]: If you want then we should put this in thread module also I saw odin lang it has an other package like sync for mutices and other synchronization mechanism

File: std/sync.oc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  
+}
+
+enum Mutex_mode{
+  NORMAL   
+  RECURSIVE
+  ERRCHECK 
+}
+
+def Mutex_mode::to_thread_api(this): i32{
```
[the remark this answers is not in the document]
Comment (iadityanath8) [reply std/sync.oc:51]: Ok i see this is some cool feature

File: std/sync.oc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+
+// Soon i'll be implementing the conditional variable in this sync package 
+namespace Condimpl{
+  
+}
+
+enum Mutex_mode{
+  NORMAL
```
[the remark this answers is not in the document]
Comment (iadityanath8) [reply std/sync.oc:46]: Ok

**Pull request closed without merging**